### PR TITLE
A collection of little fixes.

### PR DESCRIPTION
**Pull request by frankban**
Fixes https://github.com/juju/juju-gui/issues/2451 (HJC profile view).
Fixes https://github.com/juju/juju-gui/issues/2450 (deployment flow "back to canvas" button after login).
Fixes https://github.com/juju/juju-gui/issues/2445 (profile view rendering oddities).
Also fixed a couple of JavaScript type error in certain conditions (e.g. logout).

### CODE REVIEW

**Comment by frankban** [comment]
**QA** must be done in both gijoe and gisf scenarios.
For gisf, at this stage you can use `guiproxy -config 'gisf: true, jujuEnvUUID: ""'`.
Refer to the linked issues for instructions.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6773/

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js [hunk 1]
```diff
@@ -62,9 +62,9 @@ YUI.add('user-profile', function() {
     _fetchMacaroonCallback: function(error, macaroon) {
       if (error) {
         console.log(error);
-      } else {
-        this.props.storeUser('charmstore', true);
+        return;
       }
+      this.props.storeUser('charmstore', true);
```
Comment (jaycee) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: This isn't a problem with the code, but I'm curious why you prefer this structure to the if/else that was there before.

**Comment by jaycee** [comment]
QA notes:

GISF:
* Confirmed that the odd, inconsistent profile rendering is resolved
* Confirmed that going back to the canvas from DF after login is resolved

**Comment by jaycee** [comment]
QA notes:

GIJoe:

Bootstrapped with staging IdM as the identity provider. Went to /login to attempt to load /u/jcsackett; after login I had no canvas, no profile, etc.

2017-02-15-105931_1918x1042_scrot

Went to /new on same instance. URL updated to /new. Going through DF I attempted login and saw the same results as above after login.

**Comment by jaycee** [comment]
As another follow up note: I noticed that I could go to /new as an anonymous user on a GIJoe instance w/o an identity provider; there's no real logical way to then login since DF only provides the USSO thing.

Do we want to support delayed login on /new with regular credentials as well, or just not provide /new to not logged in users on Juju w/o an id provider?

**Comment by frankban** [comment]
Sending this from phone
JC anonymous should only be allowed in gisf. 
Are those two QA issues introduced by this branch?  If not, please file two bugs for a follow up?

**Comment by jaycee** [comment]
@frankban 

1. I've filed an issue for anonymous in GIJoe. https://github.com/juju/juju-gui/issues/2454
2. The issue exists in develop; I've filed bugs for them as well. https://github.com/juju/juju-gui/issues/2455

**Comment by bac** [comment]
# QA Notes II

I did QA in HJC at https://54.237.46.140/horizon/juju/gui/

Everything seems to work swimmingly well! I tried all of the URLs in the _QA 2.3.0 To Do_ document and everything worked. Hurrah!

Well, it should be noted that `/logout` doesn't work but that is expected.

**Comment by hatched** [comment]
Thanks everyone! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**